components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -0,0 +1,178 @@
+        geckoChoices: Array<out GeckoChoice>,
+        callback: ChoiceCallback
+    ) {
+
+        val pair = convertToChoices(geckoChoices)
+        // An array of all the GeckoChoices transformed as local Choices object.
+        val choices = pair.first
+        // A map that contains all local choices and map to GeckoChoices
+        val mapChoicesToGeckoChoices = pair.second
+
+        when (type) {
+
+            CHOICE_TYPE_SINGLE -> {
+                geckoEngineSession.notifyObservers {
+                    onPromptRequest(SingleChoice(choices) { selectedChoice ->
+                        val geckoChoice = mapChoicesToGeckoChoices[selectedChoice]
+                        callback.confirm(geckoChoice)
+                    })
+                }
+            }
+
+            CHOICE_TYPE_MENU -> {
+                geckoEngineSession.notifyObservers {
+                    onPromptRequest(MenuChoice(choices) { selectedChoice ->
+                        val geckoChoice = mapChoicesToGeckoChoices[selectedChoice]
+                        callback.confirm(geckoChoice)
+                    })
+                }
+            }
+
+            CHOICE_TYPE_MULTIPLE -> {
+                geckoEngineSession.notifyObservers {
+                    onPromptRequest(MultipleChoice(choices) { choices ->
+                        val ids = choices.toIdsArray()
+                        callback.confirm(ids)
+                    })
+                }
+            }
+        }
+    }
+
+    override fun onButtonPrompt(
+        session: GeckoSession?,
+        title: String?,
+        msg: String?,
+        btnMsg: Array<out String>?,
+        callback: ButtonCallback?
+    ) {
+    }
+
+    override fun onDateTimePrompt(
+        session: GeckoSession?,
+        title: String?,
+        type: Int,
+        value: String?,
+        min: String?,
+        max: String?,
+        callback: TextCallback?
+    ) {
+    }
+
+    override fun onFilePrompt(
+        session: GeckoSession?,
+        title: String?,
+        type: Int,
+        mimeTypes: Array<out String>?,
+        callback: FileCallback?
+    ) {
+    }
+
+    override fun onColorPrompt(
+        session: GeckoSession?,
+        title: String?,
+        value: String?,
+        callback: TextCallback?
+    ) {
+    }
+
+    override fun onAuthPrompt(
+        session: GeckoSession?,
+        title: String?,
+        msg: String?,
+        options: AuthOptions?,
+        callback: AuthCallback?
+    ) {
+    }
+
+    override fun onAlert(
+        session: GeckoSession?,
+        title: String?,
+        msg: String?,
+        callback: AlertCallback?
+    ) {
+    }
+
+    override fun onTextPrompt(
+        session: GeckoSession?,
+        title: String?,
+        msg: String?,
+        value: String?,
+        callback: TextCallback?
+    ) {
+    }
+
+    override fun onPopupRequest(session: GeckoSession?, targetUri: String?): GeckoResult<AllowOrDeny> {
+        TODO("not implemented")
```

Comment:
Let's not use `TODO()` because this will throw an exception when called.

Reply:
👍 I will return a dummy value!